src/composables/queries/usePoolsQuery.ts

```diff
@@ -193,19 +193,26 @@ export default function usePoolsQuery(
    */
   const queryFn = async ({ pageParam = 0 }) => {
     const fetchOptions = getFetchOptions(pageParam);
-
-    const pools: Pool[] = await poolsRepository.fetch(fetchOptions);
-
-    const skip = poolsRepository.currentProvider?.skip
-      ? poolsRepository.currentProvider.skip
-      : 0;
-
-    poolsStoreService.setPools(pools);
-
-    return {
-      pools,
-      skip,
-    };
+    let skip = 0;
+    try {
+      const pools: Pool[] = await poolsRepository.fetch(fetchOptions);
+
+      skip = poolsRepository.currentProvider?.skip
+        ? poolsRepository.currentProvider.skip
+        : 0;
+
+      poolsStoreService.setPools(pools);
+
+      return {
+        pools,
+        skip,
+      };
+    } catch (e) {
+      if (poolsStoreService.pools.value) {
+        return { pools: poolsStoreService.pools.value, skip };
```

Comment:
might be arguable

Reply:
Will it return an empty array if `poolsStoreService.pools` is empty?

Reply:
initial value here is `null`, so we don't expect there an empty array, but yeah, it's possibly better to check it for emptiness as well. will fix that